qep-345-line-symbollayer-blankareas.md

```diff
@@ -0,0 +1,117 @@
+# QGIS Enhancement: Markers/Hashes line symbol layer blank areas 
+
+**Date** 2025/07/22
+
+**Author** Julien Cabieces (@troopa81)
+
+**Contact** julien dot cabieces at oslandia dot com
+
+**Version** QGIS 4.0
+
+# Summary
+
+This proposal relates to the line symbol layer which draw markers or hashes at intervals along a line. When using such symbol layer, the display of the repeated pattern (hash or marker) can be ugly, and user would like to have a better control on what's displayed.
+
+![issue.png](images/qep345/issue.png)
+
+The rest of the proposal will mention markers but everything describe here applies also to hashes.
+
+## Proposed Solution
+
+In order to have a proper display of the markers along the line, I propose to:
+- Give user a way to identify blank area where markers are not displayed
+- Use the already existing annotation feature to add, move, rotate or even change marker symbology
+
+This could look like approximatively this
+
+![poc.gif](images/qep345/poc.gif)
+
+### Fixed marker position
+
+We also propose to add an option "fixed marker position" to fix markers positions whatever the current displayed area and zoom level. Indeed, without this option, you would get this behavior
+
+![nonfixed.gif](images/qep345/nonfixed.gif)
+
+This option will be false by default which is the actual behavior.
+
+Enabling this option will imply that we disable feature clipping for the given line symbol layer. The flag `Qgis::SymbolLayerFlag::DisableFeatureClipping` would be added to current flags if this option is set.
+
+When set, all the geometry would be drawn entirely and not clipped. This would have performance implication. A tooltip would inform user about this.
+
+### The blank areas
+
+These areas would be defined as start and end distances from the first point of the line or polygon. It would be possible to define the unit of those distances.
```

Comment:
Can you add a note something like "while it will be possible to use screen or page based units like millimeters or inches, it is expected that this will be of limited use as the blank regions would then change size when the map scale is changed. Rather, it is expected users will typically set the region distance unit to map units so that the regions remain fixed in place regardless of the map scale. Accordingly the default unit for the property will be set to Map Units."